superset/datasets/commands/samples.py

```diff
@@ -78,3 +116,24 @@ def validate(self) -> None:
             security_manager.raise_for_ownership(self._model)
         except SupersetSecurityException as ex:
             raise DatasetForbiddenError() from ex
+
+    @staticmethod
+    def get_limit_clause(
+        page: Optional[int], per_page: Optional[int]
+    ) -> Dict[str, int]:
+        samples_row_limit = app.config.get("SAMPLES_ROW_LIMIT", 1000)
+        limit = samples_row_limit
+        offset = 0
+
+        if isinstance(page, int) and isinstance(per_page, int):
+            limit = int(per_page)
+            if limit < 0 or limit > samples_row_limit:
+                # reset limit value if input is invalid
+                limit = samples_row_limit
+
+            offset = (int(page) - 1) * limit
```

Comment:
From what I've seen on the frontend, page indices start at 0.

Reply:
The page number starts from zero instead of 0.